packages/components/src/internal/components/auditlog/utils.ts

```diff
@@ -79,7 +79,7 @@ export function getEventDataValueDisplay(d: any, showLink = true): ReactNode {
             display = d;
         } else if (typeof d === 'boolean') {
             display = d ? 'true' : 'false';
-        } else {
+        } else if (d.has && d.get) {  // this is a hacky way to check if immutablejs
```

Comment:
Agreed that this is hacky. I think we can do better and expect that it must be an `Immutable.Map` then check using `Map.isMap(d)`. Here is my proposed implementation:

```ts
export function getEventDataValueDisplay(d: any, showLink = true): ReactNode {
    let display = null;
    if (d) {
        if (typeof d === 'string' || typeof d === 'number') {
            display = d;
        } else if (typeof d === 'boolean') {
            display = d ? 'true' : 'false';
        } else if (Map.isMap(d)) {
            if (d.has('formattedValue')) {
                display = d.get('formattedValue');
            } else {
                const o = d.has('displayValue') ? d.get('displayValue') : d.get('value');
                display = o?.toString() ?? null;
            }

            if (showLink) {
                if (d.get('url')) {
                    display = React.createElement('a', { href: d.get('url') }, display);
                } else {
                    const url = getTimelineEntityUrl(d.toJS());
                    if (url) {
                        display = React.createElement('a', { href: url.toHref() }, display);
                    }
                }
            }
        } else {
            if (d.formattedValue) {
                display = d.formattedValue;
            } else {
                display = (d.displayValue ?? d.value)?.toString() ?? null;
            }

            if (showLink) {
                const href = d.url ?? getTimelineEntityUrl(d)?.toHref();
                if (href) {
                    display = React.createElement('a', { href }, display);
                }
            }
        }
    }

    return display;
}
```

Here is the associated test I wrote to cover this:

```ts
test('getEventDataValueDisplay', () => {
    expect(getEventDataValueDisplay(undefined)).toEqual(null);
    expect(getEventDataValueDisplay('asAString')).toEqual('asAString');
    expect(getEventDataValueDisplay(15)).toEqual(15);
    expect(getEventDataValueDisplay(true)).toEqual('true');

    let data: any = { value: 'RawValue' };

    // Respects "value"
    expect(getEventDataValueDisplay(Map(data), false)).toEqual(data.value);
    expect(getEventDataValueDisplay(data, false)).toEqual(data.value);

    data.displayValue = 'DefaultValue';

    // Respects "displayValue"
    expect(getEventDataValueDisplay(Map(data), false)).toEqual(data.displayValue);
    expect(getEventDataValueDisplay(data, false)).toEqual(data.displayValue);

    data.formattedValue = 'NiceFormatValue';

    // Respects "formattedValue"
    expect(getEventDataValueDisplay(Map(data), false)).toEqual(data.formattedValue);
    expect(getEventDataValueDisplay(data, false)).toEqual(data.formattedValue);

    // showLink -- without a URL
    expect(getEventDataValueDisplay(Map(data), true)).toEqual(data.formattedValue);
    expect(getEventDataValueDisplay(data, true)).toEqual(data.formattedValue);

    data.urlType = SAMPLES_KEY;

    // showLink -- without a URL
    let node = getEventDataValueDisplay(data, true) as ReactElement;
    expect(node.type).toEqual('a');
    expect(node.props).toEqual({
        children: data.formattedValue,
        href: getTimelineEntityUrl(data).toHref(),
    });

    data.url = '#/some/location';

    // showLink -- with a URL
    node = getEventDataValueDisplay(Map(data), true) as ReactElement;
    expect(node.type).toEqual('a');
    expect(node.props).toEqual({ children: data.formattedValue, href: data.url });
});
```

Reply:
I was under the impression that d could be an immutable Record here.  And Record.isRecord is only supported in the next release candidate of immutableJS.  Testing this out though so far everything seems ok with this update.